client.go

```diff
@@ -607,28 +632,35 @@ func (client *client) backgroundMetadataUpdater() {
 	for {
 		select {
 		case <-ticker.C:
-			topics := []string{}
-			if !client.conf.Metadata.Full {
-				if specificTopics, err := client.Topics(); err != nil {
-					Logger.Println("Client background metadata topic load:", err)
-					break
-				} else if len(specificTopics) == 0 {
-					Logger.Println("Client background metadata update: no specific topics to update")
-					break
-				} else {
-					topics = specificTopics
-				}
-			}
-
-			if err := client.RefreshMetadata(topics...); err != nil {
-				Logger.Println("Client background metadata update:", err)
+			if err := client.refreshMetadata(); err != nil {
+				Logger.Printf("Client background metadata update:%s", err)
```

Comment:
This will not format properly (there's no space between the `:` and the error, and there's no newline at the end). You can fix the format string, but it's probably simpler to just go back to `Println` instead of `Printf` which does the right thing.